Tests/Tests/CampaignDispatcherSpec.swift

```diff
@@ -0,0 +1,333 @@
+import Quick
+import Nimble
+
+@testable import RInAppMessaging
+
+class CampaignDispatcherCarouselSpec: QuickSpec {
+    override func spec() {
+        var dispatcher: CampaignDispatcher!
+        var mockRouter: RouterMock!
+        var mockPermissionService: DisplayPermissionServiceMock!
+        var mockCampaignRepository: CampaignRepositoryMock!
+        var imageDetails: [String: ImageDetails]!
+        var images: [UIImage?]!
+        var result: [CarouselData]!
+
+        beforeEach {
+            mockRouter = RouterMock()
+            mockPermissionService = DisplayPermissionServiceMock()
+            mockCampaignRepository = CampaignRepositoryMock()
+
+            dispatcher = CampaignDispatcher(
+                router: mockRouter,
+                permissionService: mockPermissionService,
+                campaignRepository: mockCampaignRepository
+            )
+        }
+
+        afterEach {
+            MockURLProtocol.removeAllStubs()
+        }
+
+        describe("fetchImagesArray") {
+            context("when the carousel has no images") {
+                it("returns an empty array") {
+                    let carousel = Carousel(images: nil)
+
+                    waitUntil { done in
+                        dispatcher.fetchImagesArray(from: carousel) { images in
+                            expect(images).to(beEmpty())
+                            done()
+                        }
+                    }
+                }
+            }
+
+            context("when the carousel has valid image URLs") {
+                it("fetches and returns an array of images") {
+                    let imageDetails: [String: ImageDetails] = [
+                        "1": ImageDetails(imgUrl: "https://static.id.rakuten.co.jp/static/com/img/id/Rakuten_pc_20px@2x.png", link: "", altText: ""),
+                        "2": ImageDetails(imgUrl: "https://static.id.rakuten.co.jp/static/com/img/id/Rakuten_pc_20px@2x.png", link: "", altText: "")
+                    ]
+                    let carousel = Carousel(images: imageDetails)
+                    MockURLProtocol.stubImageRequests()
```

Comment:
What is the purpose of this? How to ensure this mock data was stubbed & being used?